.github/workflows/test.yml

```diff
@@ -32,3 +32,27 @@ jobs:
           - 3995:3995
         env:
           GREENMAIL_OPTS: "-Dgreenmail.setup.test.all -Dgreenmail.hostname=0.0.0.0 -Dgreenmail.auth.disabled -Dgreenmail.verbose"
+  test_cyrus:
+    runs-on: ubuntu-latest
+    name: Run tests with cyrus ${{ matrix.toolchain }}
+    strategy:
+      matrix:
+        toolchain: [stable, beta, nightly]
+    steps:
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+      - uses: actions/checkout@v2
+      - name: cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+    services:
+      cyrus_imapd:
+        image: outoforder/cyrus-imapd-tester:latest
```

Comment:
One worry I have with this is that version of Cyrus is likely to get out of date over time unless you commit to updating it. Is there not a standard Cyrus Docker image that's regularly maintained we can use?

Reply:
No, there is no official cyrus docker image at all.  (I had looked for that first when setting up testing for the app I'm building).   But this docker image is based on Rocky8 (derived from RHEL8) and thus that base image will be maintained for quite a while.  And I'll setup a github repo here on the next few days with the docker file.   And in reality the supported features of cyrus for IMAP haven't changed much in years.  (e.g. this is version 3.0.7 and my other servers are running 2.4.x and the functionality is the same between them). 

But as I'm maintaining several mail servers (on Cent 6, and moving toward Rocky 8 or Rocky 9) and writing admin software that is utilizing your crate, I'll be maintaining this for a while.

What I'll do next is setup a github repo with the docker image(s) and set it up so it'll pop out a Rocky 8 and Rocky 9 base image with the same setup.